### PR TITLE
migrate xml to json config

### PR DESCRIPTION
**Pull request by zyclonite**
remove spring dependency which was only used to parse the xml config

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -29,4 +29,5 @@ release.properties
 dependency-reduced-pom.xml
 buildNumber.properties
 .mvn/timing.properties
-ignite
+/ignite
+src/main/generated
```
Comment (tsegismont) [inline .gitignore:33]:
```suggestion
```

This one is not ignore in any Vert.x stack project

**Comment by zyclonite** [comment]
that was why i asked first before creating this PR...

maintaining only the vert.x needed parts of the options seems more reasonable then having too many dependencies which could cause troubles.
i could leave the part looking for xml configs in, so there would be still the option to migrate without loosing that feature.

**Comment by tsegismont** [comment]
>that was why i asked first before creating this PR...

I better understand your question now, as I said I thought the JSON config support was built-in. Sorry for the confusion.

>maintaining only the vert.x needed parts of the options seems more reasonable then having too many dependencies which could cause troubles.

Indeed, they could cause troubles, but I haven't seen any report about it so far.

To me the issue is that if Ignite changes its config API between versions, our setup could be broken.

> i could leave the part looking for xml configs in, so there would be still the option to migrate without loosing that feature.

That would help mitigate the issue :+1: 

Could you add a test to verify XML config works too? Thanks

**Comment by zyclonite** [comment]
need to add some code parts back, then i can do that

i would as well add the spring dependency for test only then

**Comment by zyclonite** [comment]
config lookup order is now ignite.xml then ignite.json then default-ignite.json

**Comment by tsegismont** [comment]
@zyclonite can you check the [build logs](https://travis-ci.org/github/vert-x3/vertx-ignite/builds/712675281?utm_source=github_status&utm_medium=notification)? There seems to be an issue.

File: src/main/asciidoc/index.adoc [hunk 2]
```diff
@@ -78,15 +78,18 @@ https://apacheignite.readme.io/docs/page-memory[Ignite Virtual Memory] article.
 
 === Using configuration file
 
-The cluster manager is configured by a file `default-ignite.xml` which is packaged inside the jar.
+The cluster manager is configured by a file `default-ignite.json` which is packaged inside the jar.
```
Comment (tsegismont) [inline src/main/asciidoc/index.adoc:81]: Please add a link to `IgniteOptions` and explain that the JSON content maps this data object.

File: src/main/asciidoc/index.adoc [hunk 3]
```diff
@@ -78,15 +78,18 @@ https://apacheignite.readme.io/docs/page-memory[Ignite Virtual Memory] article.
...
 
-If you want to override this configuration you can provide `ignite.xml` file on your classpath and this will be
+If you want to override this configuration you can provide `ignite.json` file on your classpath and this will be
 used instead.
 
-The xml file is a Ignite configuration file and is described in details in
+If you want to fallback to the original xml configuration you can provide `ignite.xml` file on your classpath
```
Comment (tsegismont) [inline src/main/asciidoc/index.adoc:86]: New users don't know about the original xml configuration. Can you please rephrase to say that alternatively, an XML file can be used, with a small example

File: src/main/asciidoc/index.adoc (hunk 2 above)
Comment (tsegismont) [reply src/main/asciidoc/index.adoc:81]: Also, you could add a small json example

File: src/main/asciidoc/index.adoc [hunk 4]
```diff
@@ -78,15 +78,18 @@ https://apacheignite.readme.io/docs/page-memory[Ignite Virtual Memory] article.
...
+and this will be used instead. Note: You need to add back the `ignite-spring` module to your classpath.
```
Comment (tsegismont) [inline src/main/asciidoc/index.adoc:87]: Can you add dependency details for Maven and Gradle, as done at the top of this doc for `vertx-ignite`?

**Comment by zyclonite** [comment]
@tsegismont no problem, hope you've had a great time off

**Comment by zyclonite** [comment]
@tsegismont did you find some time to review my changes?

File: src/main/asciidoc/index.adoc [hunk 5]
```diff
@@ -81,10 +81,127 @@ https://apacheignite.readme.io/docs/page-memory[Ignite Virtual Memory] article.
 The cluster manager is configured by a file `default-ignite.json` which is packaged inside the jar.
 
 If you want to override this configuration you can provide `ignite.json` file on your classpath and this will be
-used instead.
+used instead. The config maps to {@link io.vertx.spi.cluster.ignite.IgniteOptions} where you can find more details on
+each individual option.
 
-If you want to fallback to the original xml configuration you can provide `ignite.xml` file on your classpath
-and this will be used instead. Note: You need to add back the `ignite-spring` module to your classpath.
+In the example below the default config is extended to activate tls for cluster communication.
```
Comment (tsegismont) [inline src/main/asciidoc/index.adoc:87]:
```suggestion
In the example below the default config is extended to activate TLS for cluster communication.
```

File: src/main/asciidoc/index.adoc [hunk 6]
```diff
@@ -81,10 +81,127 @@ https://apacheignite.readme.io/docs/page-memory[Ignite Virtual Memory] article.
...
+    "trustStoreFilePath": "server.jks",
+    "trustStorePassword": "changeme",
+    "trustAll": false
+  },
+  "includeEventTypes": ["EVT_CACHE_OBJECT_PUT", "EVT_CACHE_OBJECT_REMOVED"],
+  "metricsLogFrequency": 0
+}
+----
+
+As an alternative to the json format you can use the native ignite xml configuration. You can provide `ignite.xml` file
```
Comment (tsegismont) [inline src/main/asciidoc/index.adoc:118]:
```suggestion
As an alternative to the json format you can use the native Ignite XML configuration. You can provide an `ignite.xml` file
```

File: src/main/asciidoc/index.adoc [hunk 7]
```diff
@@ -81,10 +81,127 @@ https://apacheignite.readme.io/docs/page-memory[Ignite Virtual Memory] article.
...
+on your classpath and it will be used instead. Note: You need to add back the `ignite-spring` module to your classpath.
```
Comment (tsegismont) [inline src/main/asciidoc/index.adoc:119]:
```suggestion
on your classpath and it will be used instead.
```

File: src/main/asciidoc/index.adoc [hunk 8]
```diff
@@ -81,10 +81,127 @@ https://apacheignite.readme.io/docs/page-memory[Ignite Virtual Memory] article.
...
+
+Add the ignite-spring dependency.
```
Comment (tsegismont) [inline src/main/asciidoc/index.adoc:121]:
```suggestion
First, add the `ignite-spring` dependency.
```

File: src/main/asciidoc/index.adoc [hunk 9]
```diff
@@ -81,10 +81,127 @@ https://apacheignite.readme.io/docs/page-memory[Ignite Virtual Memory] article.
...
+----
+
+* Gradle (in your `build.gradle` file):
+
+[source,groovy,subs="+attributes"]
+----
+compile 'org.apache.ignite:ignite-spring:${ignite.version}'
+----
+
+* Example for a default xml config:
```
Comment (tsegismont) [inline src/main/asciidoc/index.adoc:141]:
```suggestion
Then add an `ignite.xml` file like this one:
```

**Review by tsegismont — APPROVED** [review]
@zyclonite thanks for the updates. Approved with a few suggestions.

@agura can you please take a look?

**Comment by tsegismont** [comment]
Thank you @zyclonite !

**Pull request merged**